libs/langchain/langchain/vectorstores/tiledb.py

```diff
@@ -0,0 +1,383 @@
+
+        Args:
+            embedding: Embedding to look up documents similar to.
+            k: Number of Documents to return. Defaults to 4.
+            search_k: inspect up to search_k nodes which defaults
+                to n_trees * n if not provided
+
+        Returns:
+            List of Documents most similar to the embedding.
+        """
+        docs_and_scores = self.similarity_search_with_score_by_vector(
+            embedding, k, search_k
+        )
+        return [doc for doc, _ in docs_and_scores]
+
+    def similarity_search_by_index(
+        self, docstore_index: int, k: int = 4, search_k: int = -1, **kwargs: Any
+    ) -> List[Document]:
+        """Return docs most similar to docstore_index.
+
+        Args:
+            docstore_index: Index of document in docstore
+            k: Number of Documents to return. Defaults to 4.
+            search_k: inspect up to search_k nodes which defaults
+                to n_trees * n if not provided
+
+        Returns:
+            List of Documents most similar to the embedding.
+        """
+        docs_and_scores = self.similarity_search_with_score_by_index(
+            docstore_index, k, search_k
+        )
+        return [doc for doc, _ in docs_and_scores]
+
+    def similarity_search(
+        self, query: str, k: int = 4, search_k: int = -1, **kwargs: Any
+    ) -> List[Document]:
+        """Return docs most similar to query.
+
+        Args:
+            query: Text to look up documents similar to.
+            k: Number of Documents to return. Defaults to 4.
+            search_k: inspect up to search_k nodes which defaults
+                to n_trees * n if not provided
+
+        Returns:
+            List of Documents most similar to the query.
+        """
+        docs_and_scores = self.similarity_search_with_score(query, k, search_k)
+        return [doc for doc, _ in docs_and_scores]
+
+    @classmethod
+    def __from(
+        cls,
+        texts: List[str],
+        embeddings: List[List[float]],
+        embedding: Embeddings,
+        metadatas: Optional[List[dict]] = None,
+        metric: str = DEFAULT_METRIC,
+        array_uri: str = "/tmp/tiledb_array",
+    ) -> TileDB:
+        if metric not in INDEX_METRICS:
+            raise ValueError(
+                (
+                    f"Unsupported distance metric: {metric}. "
+                    f"Expected one of {list(INDEX_METRICS)}"
+                )
+            )
+        tiledb_vs = dependable_tiledb_import()
+        if not embeddings:
+            raise ValueError("embeddings must be provided to build a TileDB index")
+
+        try:
+            tiledb.group_create(array_uri)
+        except tiledb.TileDBError as err:
+            raise err
+        group = tiledb.Group(array_uri, "w")
+        vector_array_uri = f"{group.uri}/{VECTOR_ARRAY_NAME}"
+        docs_uri = f"{group.uri}/{DOCUMENTS_ARRAY_NAME}"
+        source_uri = f"/tmp/data-{random.randint(0, 999999)}"
```

Comment:
We should use `tempfile.mkdtemp` here, because this will not work on Windows and might break elsewhere w/ eg restricted `/tmp` permissions.